src/components/smart/DeploymentDetailsPage/DeploymentDetailsPage.tsx

```diff
@@ -250,14 +250,52 @@ class DeploymentDetailsPage extends React.Component<Props, State> {
       },
     );
 
-    instanceStore.loadInstancesDetails({
-      endpointId: details.origin_endpoint_id,
-      instances: details.instances.map(n => ({ id: n })),
-      cache: options.cache,
-      quietError: false,
-      env: details.source_environment,
-      targetProvider: targetEndpoint?.type,
+    const isDeploymentCompleted = this.isDeploymentCompleted();
+    const hasStoredVmInfo =
+      details.info && Object.keys(details.info).length > 0;
+
+    if (isDeploymentCompleted && hasStoredVmInfo) {
+      this.populateInstanceStoreFromDeploymentInfo(details.info);
+    } else {
+      instanceStore.loadInstancesDetails({
+        endpointId: details.origin_endpoint_id,
+        instances: details.instances.map(n => ({ id: n })),
+        cache: options.cache,
+        quietError: false,
+        env: details.source_environment,
+        targetProvider: targetEndpoint?.type,
+      });
+    }
+  }
+
+  isDeploymentCompleted(): boolean {
+    const status = this.getStatus();
+    const completedStatuses = ["COMPLETED", "ERROR", "CANCELLED"];
+    return status ? completedStatuses.includes(status) : false;
+  }
+
+  populateInstanceStoreFromDeploymentInfo(deploymentInfo: any) {
+    if (!deploymentInfo) return;
+
+    const instancesDetails = Object.keys(deploymentInfo).map(vmName => {
+      const vmData = deploymentInfo[vmName];
+      const exportInfo = vmData.export_info || {};
+
+      return {
+        id: exportInfo.id || vmName,
+        name: exportInfo.name || vmName,
+        instance_name: exportInfo.instance_name || vmName,
+        firmware_type: exportInfo.firmware_type,
+        num_cpu: exportInfo.num_cpu,
+        memory_mb: exportInfo.memory_mb,
+        os_type: exportInfo.os_type,
+        flavor_name: exportInfo.flavor_name,
+        devices: exportInfo.devices || {},
+      };
     });
+
+    instanceStore.instancesDetails = instancesDetails;
```

Comment:
I'd prefer it if we could directly print these details out instead of overwriting the `instanceStore`